bot.py

```diff
@@ -112,13 +124,119 @@ def on_message(ws, message):
+            add_user(response)
+            add_channel(response)
+            engine.execute("INSERT IGNORE INTO emojis (name, is_custom) VALUES (%s, 0)", reaction)
+            engine.execute("INSERT IGNORE INTO emojiActivity (from_user_id, to_user_id, in_channel_id, emoji_name, given_count) VALUES(%s, %s, %s, %s, 0)", response["user"], response["item_user"], response["item"]["channel"], reaction)
+            engine.execute("UPDATE emojiActivity SET given_count = given_count-1 WHERE from_user_id = %s and to_user_id = %s and in_channel_id = %s and emoji_name = %s", response["user"], response["item_user"], response["item"]["channel"], reaction)
+        return
+
+    if response["type"] == "channel_created":
+        engine.execute("INSERT IGNORE INTO channels (slack_id, name, is_productive, is_active) VALUES (%s, %s, 0, 1)", response["channel"]["id"], response["channel"]["name"])
+        return
+
+    if response["type"] == "channel_archive":
+        add_channel(response)
+        engine.execute("UPDATE channels SET is_active = 0 WHERE slack_id = %s", response["channel"])
+        return
+
+    if response["type"] == "channel_unarchive":
+        add_channel(response)
+        engine.execute("UPDATE channels SET is_active = 1 WHERE slack_id = %s", response["channel"])
+        return
+
+def setup_tables():
+    global engine
+    try:
+        engine.connect()
+    except Exception:
+        print 'Creating the PB database'
+        engine = create_engine('mysql://root@localhost:3306')
+        engine.execute("CREATE DATABASE pantherbot_test")
+        engine.execute("USE pantherbot_test")
+        engine.execute("CREATE TABLE channels(slack_id VARCHAR(9), name VARCHAR (50), is_productive TINYINT, is_active TINYINT, PRIMARY KEY (slack_id))")
+        engine.execute("CREATE TABLE users(slack_id VARCHAR(9), first_name VARCHAR(40), last_name VARCHAR(40), is_admin TINYINT, is_pb_admin, PRIMARY KEY (slack_id))")
+        engine.execute("CREATE TABLE emojis(name VARCHAR(60), is_custom TINYINT, PRIMARY KEY (name))")
+        engine.execute("CREATE TABLE commentActivity(from_user_id VARCHAR(9), to_channel_id VARCHAR(9), comment_count INTEGER, PRIMARY KEY (from_user_id, to_channel_id), FOREIGN KEY (from_user_id) REFERENCES users (slack_id), FOREIGN KEY (to_channel_id) REFERENCES channels (slack_id))")
+        engine.execute("CREATE TABLE emojiActivity(from_user_id VARCHAR(9), to_user_id VARCHAR(9), in_channel_id VARCHAR(9), emoji_name VARCHAR(60), given_count INTEGER, PRIMARY KEY (from_user_id, to_user_id, in_channel_id, emoji_name),  FOREIGN KEY (from_user_id) REFERENCES users (slack_id), FOREIGN KEY (to_user_id) REFERENCES users (slack_id), FOREIGN KEY (in_channel_id) REFERENCES channels (slack_id))")    
+        engine.execute("CREATE TABLE channelActivity(channel_id VARCHAR(9), hour TINYINT, week_day VARCHAR(2), day_of_month TINYINT, month TINYINT, year SMALLINT, FOREIGN KEY (channel_id) REFERENCES channels (slack_id))")
+
+setup_tables()
```

Comment:
Ah I see what you're doing, i'm pretty sure sqlalchemy has a built in way of handling this.